lib/external-housing-available/external-housing-available.controllers.js

```diff
@@ -175,9 +181,67 @@ module.exports = (models) => {
       return h
         .response(externalHousingAvailable);
     }, 
+    //Get ALL external housing available
+    getAllExternalHousingAvailable: async function(request, h) {
+      let returnAllExternalHousingAvailable;
+      try {
+        returnAllExternalHousingAvailable = await getAllExternalHousingAvailable(request.query);
+      } catch (e) {
+        throw Boom.badImplementation(`Error during getAllExternalHousingAvailable(). ${e}`);
+      }
+      return h
+        .response(returnAllExternalHousingAvailable);
+    },
   };
 
   function createExternalHousingAvailable(externalHousingAvailableObject) {
     return models.ExternalHousingAvailable.create(externalHousingAvailableObject);
   };
+
+  function getAllExternalHousingAvailable(queryParams = {}) {
+    return models.ExternalHousingAvailable.findAll(getAllExternalHousingAvailableOptions(queryParams));
+  }
+
+  function getAllExternalHousingAvailableOptions(queryParams = {}) {
+    let whereParams = {};
+    let externalHousingAvailableOptions = {};
+    
+    if (Object.prototype.hasOwnProperty.call(queryParams, 'title')) {
+      whereParams.title = {
+        [Op.iLike]: '%' + queryParams.title + '%',
+      }
+    }
+    if (Object.prototype.hasOwnProperty.call(queryParams, 'address')) {
+      whereParams.address = {
+        [Op.iLike]: '%' + queryParams.address + '%',
+      };
+    } 
+    if (Object.prototype.hasOwnProperty.call(queryParams, 'bedrooms')) {
+      whereParams.bedrooms = {
+        [Op.gte]: queryParams.bedrooms,
+      };
+    }
+    if (Object.prototype.hasOwnProperty.call(queryParams, 'bathrooms')) {
+      whereParams.bathrooms = {
+        [Op.gte]: queryParams.bathrooms,
+      };
+    }
+    if (Object.prototype.hasOwnProperty.call(queryParams, 'status')) {
+      whereParams.status = {
+        [Op.iLike]: '%' + queryParams.status + '%',
+      };
+    }
+    if (Object.prototype.hasOwnProperty.call(queryParams, 'price')) {
+      whereParams.price = {
+        [Op.lte]: queryParams.price,
+      };
+    }
+    if (Object.prototype.hasOwnProperty.call(queryParams, 'locations')) {
+      whereParams.LocationId = queryParams.locations;
```

Comment:
hmm. this is how I do it in property filters and it seems to work. how does this come through when it's 1 vs 2 items though? is it string and array? I know this is how I do it in property, I just don't understand how it's working :P

Reply:
Sorry, I don't understand the question

Reply:
To my understanding, based on the location type it is an array but I dont think I am understanding what the user could enter in. Are they expected to enter the name of the location (example like Allentown and it search for the id of that name in location's db using the id.) and it returns all the housing from that location? If so does that mean we have to modify this structure in querying a name instead of id's? Thank you Zac!